content/en/scale-agent/concepts/dynamic-accounts.md

```diff
@@ -0,0 +1,139 @@
+- **Endpoint**: the URL segment after the Clouddriver root
+- **Migrate an account**: move a Clouddriver-managed account to the Scale Agent for management
+- **Request**: an instruction that isn’t fulfilled immediately and can have different outcomes; a request can be done through HTTP by the admin or internally by one of the services.
+
+## Architecture
+
+The Scale Agent stores account data in a dedicated table called `clouddriver.kubesvc_accounts`. It does not modify or delete the account data in the `clouddriver.accounts` table.
+
+An account has the following lifecycle states:
+
+- Non-transient:
+
+  - `INACTIVE`: This is the initial state. Scale Agent has provisioned the account, but the account is waiting for a migration operation.
+  - `ACTIVE`: Scale Agent watches and manages the account.
+  - `FAILED`: Scale Agent failed to add or delete an account.
+  - `ORPHANED`: Neither Scale Agent nor Clouddriver is managing the account. It is inactive. You usually see this state when restarting or bringing down all of the replicas managing that account.
+
+- Transient
+
+  - `TO_MIGRATE`: The account is waiting for migration.
+  - `ACTIVATING`: The account transferred to Scale Agent for activation.
+  - `TO_DEACTIVATE`: Indicates there is an instruction to deactivate the account.
+  - `DEACTIVATING`: A request to stop watching the account has been sent to the Scale Agent.
+
+
+### Manual migration flow
+
+Migration of an account is the combination of taking the snapshot from a credential source and then activating the accounts.
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant Plugin
+    participant Service
+
+    User->>Plugin: POST /armory/accounts
+    Plugin->>Plugin: Store in clouddriver.kubesvc_accounts
+    User->>Plugin: PATCH /armory/accounts
+    Plugin->>Service: gRPC AddAccounts
+    Service-->>Plugin: return
+    Plugin->>Plugin: Update status in clouddriver.kubesvc_accounts
+```
+
+| POST | PATCH |
+|:------|:-------|
+| {{< include "scale-agent/curl/post-armory-accounts" >}} | {{< include "scale-agent/curl/patch-armory-accounts" >}}   |
+
+See {{< linkWithTitle "scale-agent/tasks/dynamic-accounts/migrate-accounts.md" >}} for detailed instructions and examples.
+
+**What happens when you initiate a migration request**
+
+* If you **do not** include a `zoneId`, the plugin sends the request to every Clouddriver instance that has a connected Scale Agent service. Each Clouddriver instance subsequently sends the request to all of its connected Scale Agent services in an attempt to find one that can process the request.  The account's `zoneId` is updated to be the one of the Agent service that is able to process the add request for that account.
+* If you **do** provide a `zoneId`, the plugin forwards the request to only those Clouddriver instances with a matching Agent service.
+* The plugin decrypts the `kubeconfig` secret and encodes it in Basee64. Then the plugin adds the content as an attribute of the account.
```

Comment:
Base64. Let's add "before sending it to agent", and a disclaimer "the data is kept encrypted in the database".